composite/token-allocation/src/config.ts

```diff
@@ -21,24 +15,47 @@ enum DataProvider {
   Kaiko = 'kaiko',
 }
 
-const providers: Record<string, PriceAdapter> = {
-  [DataProvider.Amberdata]: amberdata,
-  [DataProvider.Cryptocompare]: cryptocompare,
-  [DataProvider.Coinpaprika]: coinpaprika,
-  [DataProvider.Nomics]: nomics,
-  [DataProvider.Coinmarketcap]: coinmarketcap,
-  [DataProvider.Coingecko]: coingecko,
-  [DataProvider.Coinapi]: coinapi,
-  [DataProvider.Kaiko]: kaiko,
+const providers: Record<string, DataProviderConfig> = {
+  [DataProvider.Amberdata]: {
+    batchingSupport: false,
+  },
+  [DataProvider.Cryptocompare]: {
+    batchingSupport: true,
+    batchEndpoint: 'multi',
+  },
+  [DataProvider.Coinpaprika]: {
+    batchingSupport: true,
+    batchEndpoint: 'multi',
+  },
+  [DataProvider.Nomics]: {
+    batchingSupport: true,
+  },
+  [DataProvider.Coinmarketcap]: {
+    batchingSupport: true,
+  },
+  [DataProvider.Coingecko]: {
+    batchingSupport: true,
+  },
+  [DataProvider.Coinapi]: {
+    batchingSupport: false,
+  },
+  [DataProvider.Kaiko]: {
+    batchingSupport: false,
+  },
 }
 
 export const DEFAULT_TOKEN_DECIMALS = 18
 export const DEFAULT_TOKEN_BALANCE = 1
 
 export const makeConfig = (prefix = '', provider = ''): Config => {
   const dataProvider = provider || util.getRequiredEnv('DATA_PROVIDER', prefix)
+  const dataProviderUrl = util.getRequiredEnv('DATA_PROVIDER_URL', prefix)
```

Comment:
Until we have the service discovery, adapter url will have to be set through env variable